sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -107,6 +108,7 @@ void gnrc_lorawan_reset(gnrc_lorawan_t *mac)
 
     dev->driver->set(dev, NETOPT_RX_TIMEOUT, &rx_timeout, sizeof(rx_timeout));
 
+    mac->last_dr = CONFIG_LORAMAC_DEFAULT_DR;
```

Comment:
what's the reason of this? This function only sets the RX2 frequency. `last_dr` is used to store the last sent dr.

Reply:
It is also the DR used for the uplink. See [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c#L320).

Maybe we should rename it as just dr ?

Reply:
Again the purpose here was to make sure that the DR is changed to default value in the event that there is a `gnrc_lorawan_reset(gnrc_lorawan_t *mac)` which was previously configured via ADR.

Reply:
ok, I see

Reply:
wait a moment... if `gnrc_lorawan_reset` is called, this means it doesn't matter if it was previously configured by ADR

Reply:
> wait a moment... if `gnrc_lorawan_reset` is called, this means it doesn't matter if it was previously configured by ADR

That was the case before now since we have introduced a new bool symbol, it's not the main issue now.

The problem is  this line -464 in gnrc_lorawan_mcps.
```

    if(!mac->mlme.adr){
        mac->last_dr = mcps_request->data.dr;
    }
```
If ADR is enabled  DR should not be configured. Still it need a default value to start with.

Reply:
ADR doesn't need a default value to start with Even more, why would a function that sets the `rx2_dr` would change the last DR at all? this makes no sense to me...

Reply:
`mac->last_dr ` gets the data from `mcps_request->data.dr;` , in case there is no ADR. If there is ADR, the value cannot be set by upper layer still for the first uplink there needs to be a default value.